internal/pkg/otel/run.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package otel
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/confmap/converter/expandconverter"
+	"go.opentelemetry.io/collector/otelcol"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/errors"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/release"
+)
+
+const buildDescription = "Elastic opentelemetry-collector distribution"
+
+func IsOtelConfig(ctx context.Context, pathConfigFile string) (bool, error) {
+	fileName := filepath.Base(pathConfigFile)
+	cleanFileName := strings.TrimSpace(strings.ToLower(strings.TrimSuffix(fileName, filepath.Ext(fileName))))
+	if cleanFileName == "otel" || cleanFileName == "otlp" || cleanFileName == "otelcol" {
+		return true, nil
+	}
+
+	rawConfig, err := config.LoadFile(pathConfigFile)
+	if err != nil {
+		return false, errors.New(err,
+			fmt.Sprintf("could not read configuration file %s", pathConfigFile),
+			errors.TypeFilesystem,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	mapConfig, err := rawConfig.ToMapStr()
+	if err != nil {
+		return false, errors.New(err,
+			errors.TypeConfig,
+			errors.M(errors.MetaKeyPath, pathConfigFile))
+	}
+
+	// contains agent definition
+	_, hasAgent := mapConfig["agent"]
+	_, hasInputs := mapConfig["inputs"]
+	_, hasOutputs := mapConfig["outputs"]
+
+	if hasAgent || hasInputs || hasOutputs {
+		return false, nil
+	}
+
+	// contains otel service definition
+	_, hasService := mapConfig["service"]
+	_, hasExporters := mapConfig["exporters"]
+	_, hasReceivers := mapConfig["receivers"]
+	_, hasProcessors := mapConfig["processors"]
```

Comment:
Should [`extensions`](https://opentelemetry.io/docs/collector/configuration/#extensions) be added to this list as well?